### PR TITLE
David/fix fittings

### PR DESCRIPTION
**Pull request by DavidSarlle**
This PR contains fixes for:

- Some fitting methods
- Some Plotting methods
- Plot units
- Calibrate qubit states
- allXY

All methods has been tested with qibolab/transpilers5q and the new runcard for 5q device. 

Please @andrea-pasquale can you reivew it?

### CODE REVIEW

**Review by andrea-pasquale — COMMENTED** [review]
Thanks for this fixes @DavidSarlle.
You can find some comments below.
Two more things:

- `calibrate_qubit_states_binning` should be removed from `actions_qq_5q.yml`.
https://github.com/qiboteam/qibocal/blob/451a60b04f48aa34d4fc255499752016f04f9add/runcards/actions_qq_5q.yml#L140
- Always in `actions_qq_5q.yml` in `rabi_pulse_amplitude` I suggest to change the value of `pulse_amplitude_start` 
https://github.com/qiboteam/qibocal/blob/0784cae1c247f25e2143dba693d4452288f61933/runcards/actions_qq_5q.yml#L88
if you put 0 `qq-live` will generate some warnings when the routine starts but the plot is still working. I think that it does not plot the first points because the amplitude is too low.

File: src/qibocal/calibrations/characterization/allXY.py [hunk 1]
```diff
@@ -65,28 +78,21 @@ def allXY(
...
+            prob = (2 * prob) - 1
 
             results = {
                 "MSR[V]": msr,
                 "i[V]": i,
                 "q[V]": q,
                 "phase[rad]": phase,
-                "probability[dimensionless]": prob,
+                "probability[dimensionless]": np.array(prob),
                 "gateNumber[dimensionless]": np.array(gateNumber),
```
[the remark this answers is not in the document]
Comment (DavidSarlle) [reply src/qibocal/calibrations/characterization/allXY.py:96]: removed all cast np.array and tested. It works.

File: src/qibocal/calibrations/characterization/allXY.py [hunk 2]
```diff
@@ -333,4 +355,6 @@ def _get_sequence_from_gate_pair(platform, gates, qubit, beta_param):
 
     # RO pulse starting just after pair of gates
     ro_pulse = platform.create_qubit_readout_pulse(qubit, start=sequenceDuration + 4)
+    print(ro_pulse.serial)
```
[the remark this answers is not in the document]
Comment (DavidSarlle) [reply src/qibocal/calibrations/characterization/allXY.py]: You are right. Rewmoved. Thanks

File: src/qibocal/calibrations/characterization/allXY.py [hunk 3]
```diff
@@ -175,7 +187,20 @@ def drag_pulse_tunning(
     points=10,
 ):
 
-    platform.reload_settings()
+    # platform.reload_settings()
```
[the remark this answers is not in the document]
Comment (DavidSarlle) [reply src/qibocal/calibrations/characterization/allXY.py:190]: It is correct. The platform.reload_settings() was not working properly. There is a conflict between how the action builder works with "multiqubit" actions and the platform setup. This is a temporary  fix (with the lines below "FIXME")  in order to have a working version of the allXY and calibrate_qubit_states.  

We are working on that with @aorgazf

File: src/qibocal/calibrations/characterization/allXY.py [hunk 4]
```diff
@@ -105,7 +111,20 @@ def allXY_iteration(
     software_averages=1,
     points=10,
 ):
-    platform.reload_settings()
+    # platform.reload_settings()
```
[the remark this answers is not in the document]
Comment (DavidSarlle) [reply src/qibocal/calibrations/characterization/allXY.py:114]: Same as previous comment

**Comment by DavidSarlle** [comment]
calibrate_qubit_states_binning should be removed from actions_qq_5q.yml.

Removed

**Comment by andrea-pasquale** [comment]
Thanks for the fixes @DavidSarlle.

**Comment by andrea-pasquale** [comment]
I didn't realize that this PR was pointing to `add_routines` instead of main.
I will open another PR to fix this issue.

**Comment by stavros11** [comment]
> I didn't realize that this PR was pointing to `add_routines` instead of main. I will open another PR to fix this issue.

It may be useful to remove branches that are merged to main to avoid such issues. I think if you remove `add_routines` GitHub will automatically point the open PRs to main.

**Comment by andrea-pasquale** [comment]
Yes, at the end I just merged david/fix_fittings in main to avoid creating a new PR.
You are right. I'll delete `add_routines`. Thanks for the feedback.

**Comment by scarrazza** [comment]
If needed we can enable this: https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-the-automatic-deletion-of-branches

Let me know.

**Comment by andrea-pasquale** [comment]
I think that this could a be a good idea. Feel free to enable it @scarrazza.

**Comment by scarrazza** [comment]
Done.

**Pull request merged**